File: devcamp2018/app/src/main/java/com/psato/devcamp/infrastructure/DevCampApplication.java
package com.psato.devcamp.infrastructure;

import android.app.Application;

import com.psato.devcamp.di.component.ApplicationComponent;
import com.psato.devcamp.di.module.ApplicationModule;
import com.psato.devcamp.di.component.DaggerApplicationComponent;

/**
 * Created by psato on 29/10/16.
 */

public class DevCampApplication extends Application {
    private ApplicationComponent applicationComponent;

    @Override
    public void onCreate() {
        super.onCreate();
        initializeInjector();
    }

    private void initializeInjector() {
        applicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(new ApplicationModule(this))
                .build();
    }

    public ApplicationComponent getApplicationComponent() {
        return applicationComponent;
    }
}

File: devcamp2018/app/src/main/java/com/psato/devcamp/infrastructure/DevCampApplication.kt
package com.psato.devcamp.infrastructure

import android.app.Application
import com.psato.devcamp.di.component.ApplicationComponent
import com.psato.devcamp.di.component.DaggerApplicationComponent
import com.psato.devcamp.di.module.ApplicationModule

/**
 * Created by psato on 29/10/16.
 */

class DevCampApplication : Application() {
    var applicationComponent: ApplicationComponent? = null
        private set

    override fun onCreate() {
        super.onCreate()
        initializeInjector()
    }

    private fun initializeInjector() {
        applicationComponent = DaggerApplicationComponent.builder()
                .applicationModule(ApplicationModule(this))
                .build()
    }
}
